packages/oas/src/index.ts

```diff
@@ -654,6 +654,48 @@ export default class Oas {
     return this.operation(op.url.nonNormalizedPath, method);
   }
 
+  /**
+   * Retrieve an operation in an OAS by an `operationId`.
+   *
+   * If an operation does not have an `operationId` one will be generated in place, using the
+   * default behavior of `Operation.getOperationId()`, and then asserted against your query.
+   *
+   * Note that because `operationId`s are unique that uniqueness does include casing so your the ID
+   * you are looking for will be asserted as an exact match.
+   *
+   * @see {Operation.getOperationId()}
+   * @param id The `operationId` to look up.
+   */
+  getOperationById(id: string) {
+    let found: Operation | Webhook;
+
+    Object.entries(this.getPaths()).forEach(([, operations]) => {
+      if (found) return;
```

Comment:
`Array.find()` doesn't quite do what I'm looking for here because it'll return the entire `paths` object, not the specific operation.

```
const found = Object.entries(this.getPaths()).find(([, operations]) => {
  return Object.values(operations).find(operation => operation.getOperationId() === id);
});
```

```
{
  found: [
    '/pet/{petId}',
    { get: [Operation], post: [Operation], delete: [Operation] }
  ]
}
```

Reply:
this seems to pass tests locally:

```js
found = Object.values(operations).find(operation => operation.getOperationId() === id);
```

![CleanShot 2024-08-23 at 11 31 29@2x](https://github.com/user-attachments/assets/437749f9-3a6b-4c56-a9a6-d0c82e08cec9)

Reply:
webhooks too: 

![CleanShot 2024-08-23 at 11 33 03@2x](https://github.com/user-attachments/assets/a892b144-e14a-41e8-b707-4d3a5e5aff8e)

but again it's non-blocking so your call

Reply:
pushed that up, turns out i just cant couple `find` on the `this.getPaths` line above it.

Reply:
yep makes sense!